# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1472156746 14400
#      Thu Aug 25 16:25:46 2016 -0400
# Node ID 663e30242c4b0f83f0df4dd262292aa76922956c
# Parent  88409829869c03e78517105cecf21d9e78c356a4
avoid using "expected failure" to describe known bugs

* __run_test_suite__.m: Avoid using term "expected failure" in
summary.  Point to log file for bug report info for known bugs.
* build-sparse-tests.sh, io.tst: Use "known bug" intead of "expected
failure".

diff --git a/scripts/testfun/__run_test_suite__.m b/scripts/testfun/__run_test_suite__.m
--- a/scripts/testfun/__run_test_suite__.m
+++ b/scripts/testfun/__run_test_suite__.m
@@ -79,17 +79,19 @@ function [pass, fail, xfail, skip] = __r
       endif
       if (dsk > 0)
         printf ("  SKIPPED %6d\n", dsk);
       endif
       puts ("\n");
       printf ("See the file %s for additional details.\n", logfile);
       if (dxf > 0)
         puts ("\n");
-        puts ("Expected failures (listed as XFAIL above) are known bugs.\n");
+        puts ("Items listed as XFAIL above are known bugs.\n");
+        puts ("Bug report numbers for them may be found in the log file:\n");
+        puts (logfile);
         puts ("Please help improve Octave by contributing fixes for them.\n");
       endif
       if (dsk > 0)
         puts ("\n");
         puts ("Tests are most often skipped because the features they require\n");
         puts ("have been disabled.  Features are most often disabled because\n");
         puts ("they require dependencies that were not present when Octave\n");
         puts ("was built.  The configure script should have printed a summary\n");
diff --git a/test/build-sparse-tests.sh b/test/build-sparse-tests.sh
--- a/test/build-sparse-tests.sh
+++ b/test/build-sparse-tests.sh
@@ -926,17 +926,17 @@ gen_save_tests() {
 %!testif HAVE_HDF5   # save hdf5
 %! savefile = tempname ();
 %! as_save = as;
 %! save ("-hdf5", savefile, "bf", "as_save", "af");
 %! clear as_save;
 %! load (savefile, "as_save");
 %! unlink (savefile);
 %! assert (as_save, sparse (af));
-## FIXME: We should skip (or mark as an expected failure) the test for
+## FIXME: We should skip (or mark as a known bug) the test for
 ## saving sparse matrices to MAT files when using 64-bit indexing since
 ## that is not implemented yet.
 %!test # save matlab
 %! savefile = tempname ();
 %! as_save = as;
 %! save ("-mat", savefile, "bf", "as_save", "af");
 %! clear as_save;
 %! load (savefile, "as_save");
diff --git a/test/io.tst b/test/io.tst
--- a/test/io.tst
+++ b/test/io.tst
@@ -11,17 +11,17 @@
 ## WITHOUT ANY WARRANTY; without even the implied warranty of
 ## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
-## FIXME: we should skip (or mark as an expected failure) the test for
+## FIXME: we should skip (or mark as a known bug) the test for
 ## saving sparse matrices to MAT files when using 64-bit indexing since
 ## that is not implemented yet.
 
 %!function [ret, files] = testls (input)
 %!  ## flag a1 global so as to test the storage of global flags
 %!  global a1;
 %!
 %!  ## Input or output, so as to be able to exchange between versions
